cmd/minishift/cmd/hostfolder/sftpd.go

```diff
@@ -0,0 +1,193 @@
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hostfolder
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"github.com/golang/glog"
+	"github.com/minishift/minishift/cmd/minishift/cmd/config"
+	"github.com/minishift/minishift/pkg/util/os/atexit"
+	"github.com/pkg/sftp"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"golang.org/x/crypto/ssh"
+	"io"
+	"net"
+	"os"
+)
+
+const (
+	serverPortFlag = "port"
+)
+
+var (
+	serverPort int
+
+	hostFolderSSHDCmd = &cobra.Command{
+		Use:    "sftpd",
+		Short:  "Starts sftp server on host for sshfs based host folders.",
+		Long:   `Starts sftp server on host for sshfs based host folders.`,
+		Run:    runSftp,
+		Hidden: true,
+	}
+)
+
+func init() {
+	hostFolderSSHDCmd.Flags().IntVarP(&serverPort, serverPortFlag, "p", 2022, "The server port.")
+	HostFolderCmd.AddCommand(hostFolderSSHDCmd)
+}
+
+func runSftp(cmd *cobra.Command, args []string) {
+	serverConfig := serverConfig()
+
+	port := viper.GetInt(config.HostFoldersSftpPort.Name)
+	if port == 0 {
+		port = serverPort
+	}
+
+	// Once a ServerConfig has been configured, connections can be accepted.
+	listener, err := net.Listen("tcp", fmt.Sprintf("0.0.0.0:%d", port))
```

Comment:
Can we be more restrictive than to bind on all interfaces?

Reply:
maybe, needs some doing. I'd go with this for now and create a follow up issue. There are some caveats here.